ModTek/Logging/BetterLogFileAppender.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.IO;
+using System.Linq;
+using HBS.Logging;
+using Object = UnityEngine.Object;
+
+namespace ModTek.Logging
+{
+    internal class BetterLogFileAppender : ILogAppender, IDisposable
+    {
+        private readonly StreamWriter writer;
+        private readonly BetterLogFormatter formatter;
+        private readonly string[] patternsToIgnore;
+
+        public BetterLogFileAppender(string path, BetterLogFormatter formatter = null, string[] patternsToIgnore = null)
+        {
+            this.formatter = formatter ?? new BetterLogFormatter();
+            this.patternsToIgnore = patternsToIgnore ?? new string[0];
+            writer = new StreamWriter(path) { AutoFlush = true };
+        }
+
+        public void OnLogMessage(string name, LogLevel logLevel, object message, Object context, Exception exception, IStackTrace location)
+        {
+            var line = formatter.GetFormattedLogLine(name, logLevel, message, context, exception, location);
+            if (patternsToIgnore.Any(line.Contains))
```

Comment:
FYLS uses StartsWith, but that wont work as we add the startup time by default if we compare it against the full line.

alternatively we could compare it against the message only, which should be sufficient anyway

Reply:
Does FYLS need to change in some way?

Reply:
I would like FYLS to become obsolete by this integration, so the question would be what do we need to take over that isn't yet?

Reply:
Reasonable. Quick perusal makes it look like you hit everything FYLS does, so I'm guessing nothing. :)